### PR TITLE
Input with icon design

### PR DESCRIPTION
**Pull request by toha993**

### CODE REVIEW

File: packages/react/src/input/Input.tsx [hunk 1]
```diff
@@ -27,11 +27,7 @@ export const Input = forwardRef<HTMLInputElement, InputProps>(
         w="256"
         {...ParentBoxrecipe(props)}
       >
-        {leftSection && (
-          <Box >
-            {leftSection}
-          </Box>
-        )}
```
Comment (MunifTanjim) [inline packages/react/src/input/Input.tsx]: I'm not sure what editor setup you're using... but these should never be separate commit. You should always commit linted and formatted codes.

Comment (toha993) [reply packages/react/src/input/Input.tsx]: Got it vai.

File: packages/react/src/input/Input.tsx [hunk 2]
```diff
@@ -9,16 +9,25 @@ import * as styles from "./Input.css";
...
+  } & styles.WrapperVariants
 >;
 
 export const Input = forwardRef<HTMLInputElement, InputProps>(
   (
     {
       className,
+      defaultValue,
+      id,
       isDisabled,
```
Comment (radnan) [inline packages/react/src/input/Input.tsx:26]: let's rename `isDisabled` to just `disabled` - since that's the actual attribute name. that way we're following the general convention, not making it a surprise to users, and making it consistent with other components like `Button`.

File: packages/react/src/input/Input.tsx [hunk 3]
```diff
@@ -9,16 +9,25 @@ import * as styles from "./Input.css";
...
       isInvalid,
```
Comment (radnan) [inline packages/react/src/input/Input.tsx:27]: let's rename this from `isInvalid` to `error`. there is not `error` html attribute - but this is just to remain consistent with other attributes like `disabled`, `readonly`, etc

File: packages/react/src/input/Input.tsx [hunk 4]
```diff
@@ -43,7 +51,34 @@ export const Input = forwardRef<HTMLInputElement, InputProps>(
...
+          className={clsx(
+            styles.input({
+              variant,
+            }),
+          )}
+        >
+          <input
+            defaultValue={defaultValue}
+            id={id}
+            placeholder={placeholder}
```
Comment (radnan) [inline packages/react/src/input/Input.tsx:70]: are we actually doing anything with the `placeholder`? if not let's just forward it via `restProps` instead of picking it out and passing it explicitly

File: packages/react/src/input/Input.tsx [hunk 5]
```diff
@@ -43,7 +51,34 @@ export const Input = forwardRef<HTMLInputElement, InputProps>(
...
+            {leftSection}
+          </Box>
+        )}
+        <Box
+          asChild
+          className={clsx(
+            styles.input({
+              variant,
+            }),
+          )}
```
Comment (radnan) [inline packages/react/src/input/Input.tsx:65]: let's shorten this to a single line
```suggestion
          className={styles.input({ variant })}
```

and remove `clsx` since we're not merging any classes here

File: packages/react/src/input/Input.tsx (hunk 4 above)
Comment (toha993) [reply packages/react/src/input/Input.tsx:70]: Before restProps were introduced, I needed to pass the prop explicitly for the placeholder. With restProps, we wouldn't need it anymore vai.

File: packages/react/src/input/Input.tsx (hunk 2 above)
Comment (toha993) [reply packages/react/src/input/Input.tsx:26]: Sure vai

File: packages/react/src/input/Input.tsx (hunk 3 above)
Comment (toha993) [reply packages/react/src/input/Input.tsx:27]: Okay vai

File: packages/react/src/input/Input.tsx (hunk 5 above)
Comment (toha993) [reply packages/react/src/input/Input.tsx:65]: Okay vai

**Review by radnan — APPROVED** [review]
